akka-remote/src/main/scala/akka/remote/artery/SendQueue.scala

```diff
@@ -42,26 +43,37 @@ private[remote] final class SendQueue[T] extends GraphStageWithMaterializedValue
 
   override def createLogicAndMaterializedValue(inheritedAttributes: Attributes): (GraphStageLogic, QueueValue[T]) = {
     @volatile var needWakeup = false
-    @volatile var queue: Queue[T] = null
+    @volatile var producerQueue: Queue[T] = null
```

Comment:
Good suggestion to move it to the value, belongs there. I think it still must be volatile though, since it can be different threads calling `inject` and `offer`?